### PR TITLE
Merged prefill fix - pad logits for sampler

### PR DESCRIPTION
**Pull request by adobrzyn**

### CODE REVIEW

**Comment by xuechendi** [comment]
@adobrzyn , please help to elaborate the issue and how fixing works here, thanks
